### PR TITLE
Add lowercase `reactive.value`, `calc`, `effect`

### LINKED ISSUE #814: Should add lowercase aliases for `reactive.value`, `reactive.calc`, `reactive.effect`
There are technical reasons for having the capitalized names `reactive.Value`, `reactive.Calc`, an `reactive.Effect`, but from a user perspective, it is confusing. So we should add lowercase aliases for them.

### PR DESCRIPTION
**Pull request by wch**
This closes #814.

Many users have wondered why `reactive.Value`, `.Calc`, and `.Effect` are capitalized. There are technical reasons for this, but from a user perspective it makes more sense to just use lowercase names.

This PR changes those names to lowercase, and adds aliases so that the old capitalized names still work. 

This PR does not change any examples because if users on a previous version of shiny were to use the example code, it wouldn't work for them, and it would be unclear to them why. In a future release, we should update the examples to use the new lowercase names.

### CODE REVIEW

File: shiny/reactive/__init__.py [hunk 1]
```diff
@@ -26,8 +29,11 @@
     "on_flushed",
     "poll",
     "file_reader",
+    "value",
     "Value",
+    "calc",
     "Calc",
+    "effect",
     "Effect",
```
Comment (schloerke) [inline shiny/reactive/__init__.py:37]: Should the old names be removed and pyright ignored above?

```suggestion
    "calc",
    "effect",
```

File: shiny/reactive/__init__.py [hunk 2]
```diff
@@ -8,10 +8,13 @@
...
 from ._poll import poll, file_reader
 from ._reactives import (  # noqa: F401
+    value,
     Value,
+    calc,
     Calc,
     Calc_,  # pyright: ignore[reportUnusedImport]
     CalcAsync_,  # pyright: ignore[reportUnusedImport]
+    effect,
     Effect,
```
Comment (schloerke) [inline shiny/reactive/__init__.py:18]: (Continuation of comment below)

```suggestion
    Value,  # pyright: ignore[reportUnusedImport]
    calc,
    Calc,  # pyright: ignore[reportUnusedImport]
    Calc_,  # pyright: ignore[reportUnusedImport]
    CalcAsync_,  # pyright: ignore[reportUnusedImport]
    effect,
    Effect,  # pyright: ignore[reportUnusedImport]
```

File: shiny/reactive/__init__.py (hunk 1 above)
Comment (wch) [reply shiny/reactive/__init__.py:37]: I think we need to keep the old names for backward compatibility, at least for the time being. If we were to remove them now, then a lot of user code would get flagged by pyright.

**Comment by jcheng5** [comment]
I think we have consensus that `@reactive.calc` and `@reactive.effect`, as decorators, should be lower-cased.

I'm a little surprised to see `reactive.value()` being lower-cased as well, as that's just a plain old constructor?

**Comment by jcheng5** [comment]
And IF we do really want `reactive.value` to be lower-cased, I think we should still have `class Value` and then create a `value` alias (or even make a wrapper function `def value(x: T) -> reactive.Value[T]`) so that it still makes sense to have `reactive.Value` for type annotations.

**Comment by wch** [comment]
I think we should provide a lowercase version of `value`, because if we don't, I think we'll get the same questions about "why is this thing capitalized?" But I don't feel strongly about it.

I think it makes sense to keep the `Value` class, and then make `value` an alias to it.

So in short, I can roll back the `Value`->`value` changes, and then just add `value` as an alias. Then if we decide we don't like the alias, we can take it out, and it'll be just a one-line change.

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**